docs/src/index.md

```diff
@@ -43,6 +42,16 @@ TODO: Describe access to arbitrary GAP variables and functions via `GAP.Globals.
 TODO: describe Help system integration
 
 
+## Managing GAP packages
+
+```@docs
+GAP.Packages.load
+GAP.Packages.install
+GAP.Packages.update
+GAP.Packages.remove
+```
```

Comment:
@mohamed-barakat With this you can do e.g. the following:
```
$ julia
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.3.1 (2019-12-30)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> using GAP
 ┌───────┐   GAP 4.11.0-1-ge33ffb1 of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-apple-darwin18.7.0-julia64-kv7
 Configuration:  gmp 6.1.2, Julia GC, Julia 1.3.1, readline, KernelDebug
 Loading the library and packages ...
 Packages:   GAPDoc 1.6.3, PrimGrp 3.3.2, SmallGrp 1.4.1, TransGrp 2.0.4
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'

julia> GAP.Packages.install("Browse")
#I  Getting PackageInfo URLs...
#I  Retrieving PackageInfo.g from http://www.math.rwth-aachen.de/~Browse/PackageInfo.g ...
#I  Downloading archive from URL http://www.math.rwth-aachen.de/~Browse/Browse-1.8.9.tar.bz2 ...
#I  Saved archive to /var/folders/d_/1zss2fnd6xdgclqnj8jj_27m0000gp/T//tmN90Qhv/Browse-1.8.9.tar.bz2
#I  Extracting to /Users/mhorn/.gap/pkg/Browse ...
#I  Checking dependencies for Browse...
#I    GAPDoc >= 1.6: true
#I  Running compilation script on /Users/mhorn/.gap/pkg/Browse ...
true
```

This is still a bit rough -- for example, it prints GAP info messages -- which we could of course disable by lowering the info level, but they contain helpful info which we don't have otherwise. Maybe we can work with @mtorpey on a slightly more powerful API that gives us more status information?

More importantly, there is no way to declare in your Julia package that you depend on GAP's `IO` package; so you'll have to call `GAP.Packages.install("IO")` or so; but that would always try to do something, even if "io" is already installed and compiled. So you should for now maybe also call `GAP.Globals.TestPackageAvailability("IO")`  to decide whether to call `GAP.Packages.install("IO")`.

On the long run, we could provide Julia wrapper packages for all those GAP packages; something  like `GAP-IO.jl`, `GAP-Homalg.jl`, etc. and these could model dependencies etc. as in GAP (the `Project.toml` file could be generated from the `PackageInfo.g` to that end, and the Julia package would use `GAP.LoadPackageAndExposeGlobals` or something similar. More elaborate wrappers could be created later. Each such wrapper could contain its own copy of the relevant GAP package, and add itself as a GAP root path (many variations of this scheme are of course possible).

The main advantage of this approach would be that one could declare dependencies on GAP packages like any other Julia package dependency, and then installing your package using `Pkg.add("foobar")` in Julia would automatically ensure all required GAP packages are provided. But the main (and major) drawback would be that we'd have to somehow maintain all those wrappers. The only realistic way to do IMHO would be to fully automate their generation. All in all, I am not sure this is worth the effort, but I'd like to think a bit more about this.

In the meantime, I hope at least your most immediate needs are addressed?

Reply:
I indeed am using `GAP.Globals.TestPackageAvailability("IO")` in [`CompileGapPackage`](https://github.com/homalg-project/HomalgProject.jl/blob/master/src/HomalgProject.jl#L67) used [here](https://github.com/homalg-project/HomalgProject.jl/blob/master/src/HomalgProject.jl#L42). Thanks for the hint.